cl/search/documents.py

```diff
@@ -297,3 +302,245 @@ def prepare_percolator_query(self, instance):
+        },
+        search_analyzer="search_analyzer",
+    )
+    supervisor = fields.TextField(
+        attr="supervisor.name_full_reverse",
+        analyzer="text_en_splitting_cl",
+        fields={
+            "exact": fields.TextField(
+                attr="supervisor.name_full_reverse", analyzer="english_exact"
+            ),
+        },
+        search_analyzer="search_analyzer",
+    )
+    predecessor = fields.TextField(
+        attr="predecessor.name_full_reverse",
+        analyzer="text_en_splitting_cl",
+        fields={
+            "exact": fields.TextField(
+                attr="predecessor.name_full_reverse", analyzer="english_exact"
+            ),
+        },
+        search_analyzer="search_analyzer",
+    )
+    date_nominated = fields.DateField(attr="date_nominated")
+    date_elected = fields.DateField(attr="date_elected")
+    date_recess_appointment = fields.DateField(attr="date_recess_appointment")
+    date_referred_to_judicial_committee = fields.DateField(
+        attr="date_referred_to_judicial_committee"
+    )
+    date_judicial_committee_action = fields.DateField(
+        attr="date_judicial_committee_action"
+    )
+    date_hearing = fields.DateField(attr="date_hearing")
+    date_confirmation = fields.DateField(attr="date_confirmation")
+    date_start = fields.DateField(attr="date_start")
+    date_granularity_start = fields.KeywordField(attr="date_granularity_start")
+    date_retirement = fields.DateField(attr="date_retirement")
+    date_termination = fields.DateField(attr="date_termination")
+    date_granularity_termination = fields.KeywordField(
+        attr="date_granularity_termination"
+    )
+    judicial_committee_action = fields.KeywordField()
+    nomination_process = fields.KeywordField()
+    selection_method = fields.KeywordField()
+    selection_method_id = fields.KeywordField(attr="how_selected")
+    termination_reason = fields.KeywordField()
+    text = fields.TextField(
+        analyzer="text_en_splitting_cl",
+        fields={
+            "exact": fields.TextField(analyzer="english_exact"),
+        },
+        search_analyzer="search_analyzer",
+    )
+
+    class Django:
+        model = Position
+        ignore_signals = True
+
+    def prepare_position_type(self, instance):
+        return instance.get_position_type_display()
+
+    def prepare_judicial_committee_action(self, instance):
+        return instance.get_judicial_committee_action_display()
+
+    def prepare_nomination_process(self, instance):
+        return instance.get_nomination_process_display()
+
+    def prepare_selection_method(self, instance):
+        return instance.get_how_selected_display()
+
+    def prepare_termination_reason(self, instance):
+        return instance.get_termination_reason_display()
+
+    def prepare_text(self, instance):
+        text_template = loader.get_template("indexes/person_text.txt")
+        return text_template.render({"item": instance}).translate(null_map)
```

Comment:
Can you take a look at person_text.txt and see which of those fields we actually still need? Maybe we can do without the `text` field again?

Reply:
Yeah, `text` is already removed in #2870